micrometer-core/src/main/java/io/micrometer/core/instrument/MeterRegistry.java

```diff
@@ -643,8 +656,29 @@ private Meter.Id mapId(Meter.Id id) {
         return mappedId;
     }
 
+    @FunctionalInterface
+    // Brittle, but this is internal. We pass nulls for some meter types as explained in
+    // JavaDoc.
+    @NullUnmarked
+    private interface NewMeterSupplier<M extends Meter> {
+
+        /**
+         * Create a new meter with the given parameters. The DistributionStatisticConfig
+         * and PauseDetector will be null for meter types that do not take them.
+         * @param registry the registry from which to make the new meter
+         * @param id the ID of the meter to create
+         * @param distributionStatisticConfig optional distribution config for types that
+         * take it
+         * @param pauseDetector optional pause detector for types that take it
+         * @return a new meter
+         */
+        @NonNull M create(MeterRegistry registry, Meter.Id id, DistributionStatisticConfig distributionStatisticConfig,
+                PauseDetector pauseDetector);
+
+    }
+
     private Meter getOrCreateMeter(@Nullable DistributionStatisticConfig config,
-            BiFunction<Meter.Id, /* Nullable Generic */ DistributionStatisticConfig, ? extends Meter> builder,
+            @Nullable PauseDetector specificPauseDetector, NewMeterSupplier<? extends Meter> meterSupplier,
```

Comment:
We take the pause detector as a parameter here so we don't need to capture it in the lambda for the NewMeterSupplier (previously the BiFunction).